### PR TITLE
Chore/update tooling

### PR DESCRIPTION
**Pull request by jams2**

### CODE REVIEW

File: ruff.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+line-length = 88
+target-version = "py38"
+
+[lint]
+ignore = [
+    "E501", # line too long
+    "S101", # flake8-bandit Use of assert detected
+]
+
+select = [
```
[the remark this answers is not in the document]
Comment (jams2) [reply ruff.toml]: Thanks, added in https://github.com/wagtail/wagtail-factories/pull/84/commits/3cdc8d0ad8cdfa84970d279a48850f2a1dbc71a2

File: .github/workflows/python-tox.yml [hunk 2]
```diff
@@ -5,57 +5,20 @@ on: [push, pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
-    services:
```
Comment (jams2) [inline .github/workflows/python-tox.yml:8]: I'd already changed the test settings so that sqlite in-memory is used. We don't test any specific database functionality.

File: CHANGES [hunk 3]
```diff
@@ -1,5 +1,11 @@
 Unreleased
 ==========
+- Update for Wagtail >5.1
+- Drop support for Django <4.2
```
Comment (jams2) [inline CHANGES:4]: Technically Django 3.2 has security support until April, but by the time we get to doing a release that probably won't be far off.

File: MANIFEST.in [hunk 4]
```diff
@@ -4,7 +4,7 @@ recursive-exclude * *
 
 graft src/wagtail_factories
 
-include README.rst
+include README.md
```
Comment (jams2) [inline MANIFEST.in:7]: Converted the readme to markdown, I find it more pleasant to work with than RST, and I would like there to be less obstacles to getting some documentation written.

**Review by jams2 — COMMENTED** [review]
@zerolab are you up for giving this a review?

The code and formatting changes are from ruff.

File: ruff.toml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+exclude = ["dist","build","venv",".venv",".tox",".git"]
+line-length = 88
+target-version = "py38"
+
+[lint]
+ignore = [
+    "E501", # line too long
+    "S101", # flake8-bandit Use of assert detected
```
Comment (zerolab) [inline ruff.toml:8]: it may be better to move this to https://docs.astral.sh/ruff/settings/#lint_extend-per-file-ignores and limit it to just test files.

https://github.com/wagtail/wagtail-factories/blob/b4d53cf357b6a1b96e5df68a56a79732ad7965ba/src/wagtail_factories/factories.py#L77 would be better with a proper exception

File: setup.py [hunk 6]
```diff
@@ -16,17 +16,13 @@
     "pytest-django==4.5.0",
     "pytest-cov==3.0.0",
     "pytest-pythonpath==0.7.3",
```
Comment (zerolab) [inline setup.py:18]: this may be for another PR, but pytest and the pytest plugins all have more recent releases which would be nice to use.

also pytest-pythonpath is obsolete as of pytest 7.0

File: setup.py [hunk 7]
```diff
@@ -16,17 +16,13 @@
...
-    "psycopg2>=2.3.1",
     "coverage==6.0",
```
Comment (zerolab) [inline setup.py:19]: ditto, for another PR

File: tests/settings.py [hunk 8]
```diff
@@ -7,7 +7,7 @@
 
 ROOT_URLCONF = "tests.urls"
 
-SECRET_KEY = "Gx8sMKAtnA69TR9lyAlLuSnozUv3kxdscHkpwEjatZRVQQ0laMY69KL4XPxvr3KY"
+SECRET_KEY = "Gx8sMKAtnA69TR9lyAlLuSnozUv3kxdscHkpwEjatZRVQQ0laMY69KL4XPxvr3KY"  # NOQA
```
Comment (zerolab) [inline tests/settings.py:10]:
```suggestion
SECRET_KEY = "not-a-secret-key"  # noqa: S105
```

File: tox.ini [hunk 9]
```diff
@@ -1,32 +1,32 @@
...
-    wagtail50: wagtail>=5.0,<5.1
+    django42: django>=4.2,<5.0
+    django50: django>=5.0,<5.1
+    wagtail52: wagtail>=5.2,<6.0
+    wagtail60: wagtail>=6.0,<6.1
     factoryboy32: factory-boy>=3.2,<3.3
+    factoryboy33: factory-boy>=3.3,<3.4
 
 [testenv:coverage-report]
-basepython = python3.7
```
Comment (zerolab) [inline tox.ini:29]: go with at least `3.11`. no reason not to use a faster python version

File: tox.ini [hunk 10]
```diff
@@ -35,8 +35,8 @@ commands =
...
 
 [testenv:lint]
-basepython = python3.7
+basepython = python3.8
 deps = flake8
 commands =
-    flake8 src tests setup.py
-    isort --recursive --check-only --diff src tests
+    ruff format --check src/ tests/
+    ruff check src/ tests/
```
Comment (zerolab) [inline tox.ini:42]: question: why not `.` instead of `src/ tests/`?

**Review by zerolab — CHANGES_REQUESTED** [review]
Left a few suggestions and questions

File: tox.ini (hunk 9 above)
Comment (jams2) [reply tox.ini:29]: Changed in c6dc298

File: tox.ini (hunk 10 above)
Comment (jams2) [reply tox.ini:42]: No good reason! Changed in ea8b5cc

File: ruff.toml (hunk 5 above)
Comment (jams2) [reply ruff.toml:8]: Nice suggestion, fixed in 3dba08e

File: setup.py (hunk 6 above)
Comment (jams2) [reply setup.py:18]: Thanks, opened https://github.com/wagtail/wagtail-factories/issues/86

File: setup.py (hunk 7 above)
Comment (jams2) [reply setup.py:19]: Opened https://github.com/wagtail/wagtail-factories/issues/86

**Review by zerolab — APPROVED** [review]
LGTM!
